sys/random/mersenne.c

```diff
@@ -129,7 +133,12 @@ uint32_t genrand_uint32(void)
     y ^= (y << 7) & 0x9d2c5680UL;
     y ^= (y << 15) & 0xefc60000UL;
     y ^= y >> 18;
+
+#ifdef MODULE_ENTROPY
+    return y + entropy;
```

Comment:
I felt that adding something halfway random is less predictive than setting a halfway random starting point. lets ask a mathematician, though.

-----Original Message-----
From: Martine Lenders <notifications@github.com>
To: RIOT-OS/RIOT <RIOT@noreply.github.com>
Cc: Kaspar Schleiser <kaspar@schleiser.de>
Sent: Di., 17 Nov. 2015 18:16
Subject: Re: [RIOT] sys: random: add entropy collector (#4294)

> @@ -129,7 +133,12 @@ uint32_t genrand_uint32(void)
>      y ^= (y << 7) & 0x9d2c5680UL;
>      y ^= (y << 15) & 0xefc60000UL;
>      y ^= y >> 18;
> +
> +#ifdef MODULE_ENTROPY
> +    return y + entropy;

Shouldn't the entropy introduced as a seed, rather then messing with the deterministic qualities of the PNR?

---
Reply to this email directly or view it on GitHub:
https://github.com/RIOT-OS/RIOT/pull/4294/files#r45088799